### PR TITLE
Implement DNS service discovery plugin

### PR DESCRIPTION
**Pull request by repeatedly**

### CODE REVIEW

File: fluent-plugin-sd-dns.gemspec [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  gem.version     = "0.1.0"
+  gem.authors     = ["Masahiro Nakagawa"]
+  gem.email       = "repeatedly@gmail.com"
+  #gem.platform    = Gem::Platform::RUBY
+  gem.files       = `git ls-files`.split("\n")
+  gem.test_files  = `git ls-files -- {test,spec,features}/*`.split("\n")
+  gem.executables = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
+  gem.require_paths = ['lib']
+
+  gem.add_dependency "fluentd", ">= 1"
```
Comment (cosmo0920) [inline fluent-plugin-sd-dns.gemspec]: Fluentd v1.8.0 or later is required?
https://github.com/fluent/fluentd/commit/17ae60accda16a2f133fa95ee655428964c5ea0d

File: README.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
+# DNS based service discovery plugin for [Fluentd](https://fluentd.org)
+
+The `dns` service discovery plugin updates target services by DNS list.
+
+This plugin uses `Socket.getaddrinfo` internally.
+
+## Requirements
+
+- Fluentd v1 or later
```
Comment (cosmo0920) [inline README.md]: Service discovery is supported in Fluentd v1.8.0 or later.
Should we point in accurate release version?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+```
+
+#### shared_key
+
+- Type: string
+- Default: nil
+
+The shared_key per entry
+
+#### usename
```
Comment (cosmo0920) [inline README.md]: use**r**name?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+- Type: string
+- Default: empty string
+
+The usename for authentication.
```
Comment (cosmo0920) [inline README.md]: ditto.

File: lib/fluent/plugin/sd_dns.rb [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+          queue.push(a)
+        end
+      end
+
+      def fetch_services_from_dns
+        service_list = []
+        @entry_configs.each { |conf|
+          ai_list = begin
+                      Socket.getaddrinfo(conf.host, conf.port, conf.address_family, Socket::SOCK_STREAM)
+                    rescue => e
```
Comment (cosmo0920) [inline lib/fluent/plugin/sd_dns.rb:118]: Should we rescue all exceptions or specific ones?

**Review by cosmo0920 — COMMENTED** [review]
Implementation is good for me but I added some comments. 
Could you check them?

File: lib/fluent/plugin/sd_dns.rb (hunk 5 above)
Comment (repeatedly) [reply lib/fluent/plugin/sd_dns.rb:118]: I think RuntimeException is enough. If other exceptions happens, it seems the bug of code.

Comment (cosmo0920) [reply lib/fluent/plugin/sd_dns.rb:118]: Oh, I got it. LGTM.
https://docs.ruby-lang.org/ja/2.3.0/class/StandardError.html

**Review by cosmo0920 — APPROVED** [review]

**Pull request merged**